symengine/cwrapper.cpp

```diff
@@ -399,9 +399,17 @@ void basic_pow(basic s, const basic a, const basic b)
     s->m = SymEngine::pow(a->m, b->m);
 }
 
-void basic_div(basic s, const basic a, const basic b)
+int basic_div(basic s, const basic a, const basic b)
 {
-    s->m = SymEngine::div(a->m, b->m);
+    try
+    {
+        s->m = SymEngine::div(a->m, b->m);
+        return 0;
+    }
+    catch ( ... )
```

Comment:
In SymEngine C++ layer, only std::runtime_errors are thrown, instead we can specialize std::runtime_errors and have SymEngine exceptions like `ZeroDivisionError` and in c wrapper we can return an error code for ZeroDivisionError.

In ruby you can check the error code and if it matches ZeroDivisionError you can throw a Ruby ZeroDivisionError, otherwise a generic Ruby exception